### PR TITLE
🏷️ Improve Pick Typings

### PR DESCRIPTION
**Pull request by N0tExisting**
This PR improves `src/pick.ts`'s typings.

I don't think this needs an issue

### CODE REVIEW

File: src/pick.ts [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+import type { StateTree } from 'pinia'
+
+function get(state: StateTree, path: string[]): unknown {
+  return path.reduce((obj, p) => {
+    return obj?.[p]
+  }, state)
 }
 
-function set(object: unknown, path: Array<string>, val: unknown): unknown {
+function set(state: StateTree, path: string[], val: unknown): StateTree {
```
Comment (prazdevs) [inline src/pick.ts:9]: is there a reason for `string[]` over `Array<string>` ? I always preferred using generics

Comment (N0tExisting) [reply src/pick.ts:9]: It just looks shorter. I find that easier to read.

Comment (prazdevs) [reply src/pick.ts:9]: I should have included the `array-type` rule to generic to my eslint-config 😄 
If we could stick to `Array<T>` syntax on the repo, I'd appreciate.

**Review by prazdevs — CHANGES_REQUESTED** [review]
see comment

**Pull request merged**